packages/core/src/components/region/RegionBar/RegionBar.tsx

```diff
@@ -45,9 +48,26 @@ function RegionBar({
     ButtonIcon,
   } = useOverrideComponents<'RegionBar'>()
 
-  const { openModal } = useUI()
+  const { openModal, openPopover } = useUI()
   const { city, postalCode } = useSession()
-  const shouldDisplayPostalCode = postalCode !== initialSession.postalCode
+  const { isValidationComplete } = useRegion()
+  const regionBarRef = useRef<HTMLDivElement>(null)
+
+  const defaultPostalCode =
+    !!initialSession?.postalCode && postalCode === initialSession.postalCode
+
+  // If location is not mandatory, and default zipCode is provided or if the user has not set a zipCode, show the popover.
+  const displayRegionPopover =
```

Comment:
question: should it display the popover if the `deliveryPromise.mandatory` is true?
My doubt is if this conditional should be `!deliveryPromise.mandatory && (defaultPostalCode || !postalCode)`

Reply:
good question! 
If there is no postal code set and deliveryPromise.mandatory -> the RegionModal should appear (without the close button)
if there is default postal code set and deliveryPromise.mandatory -> I think it should be the same scenario: appear the Popover, so the user can edit the postal code 🤔

Reply:
Got it! So I believe the conditional is right!

Reply:
Got it! So I believe the conditional is right!